### ISSUE TITLE
Possible invalid documentation/or functionality about check.all accepting check.map which returns object not array

### ISSUE THREAD

**Issue by idchlife** [philbooth/check-types.js#88]
You have this: https://github.com/philbooth/check-types.js#batch-operations

Which says map is for objects and returns objects.

Then you have this in the examples:

```javascript
check.all(
    check.map(
        { foo: 0, bar: '' },
        { foo: check.number, bar: check.string }
    );
);
```

Which if I understand correctly will not work, since all accepts an array, but not an object

**Comment by philbooth** [comment]
`all` and `any` are polymorphic, they work with both the arrays returned from `apply` and objects returned from `map`. It's mentioned in the docs (same section you linked to) but maybe it's not clear, I'm happy to improve them if you have suggestions.

Also fwiw, I noticed there was a typo (bad semicolon) in the example code you pasted from the readme, so I fixed that just now. Just mentioning in case you were running the above code and it was the cause of an issue.
